src/instruments/src/EFB/Navigation/Pages/LocalFilesPage/LocalFileChartSelector.tsx

```diff
@@ -59,26 +63,35 @@ export const LocalFileChartSelector = ({ selectedTab, loading }: LocalFileChartS
         );
     }
 
-    const getChartResourceUrl = async (chart: LocalFileChart) => {
-        const resp = await fetch(chart.type === 'PDF'
-            ? `http://localhost:8380/api/v1/utility/pdf?filename=${chart.fileName}&pagenumber=1`
-            : `http://localhost:8380/api/v1/utility/image?filename=${chart.fileName}`);
-
-        if (!resp.ok) {
+    const getChartResourceUrl = async (chart: LocalFileChart): Promise<string> => {
+        if (simbridgeEnabled !== 'AUTO ON') {
+            return Promise.reject();
+        }
+        try {
+            if (chart.type === 'PDF') {
+                const blob = await Viewer.getPDFPage(chart.fileName, 1);
+                return URL.createObjectURL(blob);
+            }
+            const blob = await Viewer.getImage(chart.fileName, 1);
+            return URL.createObjectURL(blob);
+        } catch (err) {
             return Promise.reject();
         }
-
-        const blob = await resp.blob();
-        return URL.createObjectURL(blob);
     };
 
     const getPagesViewable = async (chart: LocalFileChart): Promise<number> => {
+        if (simbridgeEnabled !== 'AUTO ON') {
+            return Promise.reject();
+        }
         if (chart.type === 'PDF') {
-            const pageNumResp = await fetch(`http://localhost:8380/api/v1/utility/pdf/numpages?filename=${chart.fileName}`);
-            return pageNumResp.json();
+            try {
+                const pageNumResp = await Viewer.getPDFPageNum(chart.fileName);
+                return pageNumResp.valueOf();
```

Comment:
Just a query, why use `valueOf` if the method already returns a number?

Reply:
Now I see that the state for navigationPage requires number instead of Number - so somewhere I will need to have to do the conversion or the API needs to return number instead of Number